src/views/List.jsx

```diff
@@ -2,11 +2,15 @@ import { useState, useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { ListItem } from '../components';
 
+import { comparePurchaseUrgency } from '../utils/item';
+
 export function List({ data, listToken }) {
 	const navigate = useNavigate();
 	const [searchTerm, setSearchTerm] = useState('');
 	const [copied, setCopied] = useState('');
 
+	const sortedFullList = comparePurchaseUrgency(data);
```

Comment:
One other suggestion here is that since this can be a complex calculation, it would be more performant to wrap this in a `useMemo` hook so that it doesn't get calculated during every rerender unless `data` changes.

```javascript
	const sortedFullList = useMemo(() => comparePurchaseUrgency(data), [data]);
```

Reply:
Thanks! Adding this change and learning about how to use this hook.